src/components/login/login.js

```diff
@@ -0,0 +1,61 @@
+import React, { useState } from 'react';
+import { useForm } from 'react-hook-form';
+
+import ErrorMessage from '../error-message/error-message';
+import './login.scss';
+
+function Login({ onSuccess, error }) {
+  const { register, handleSubmit, errors } = useForm();
+  const handleLogin = (data) => {
+    onSuccess(data.email, data.password);
+  };
+
+  return (
+    <form className="login" onSubmit={handleSubmit(handleLogin)} noValidate>
+      <h1 className="loginTitle">Login</h1>
+      {error ? (
+        <div className="globalError">
+          <ErrorMessage error={{ message: error }} />
+        </div>
+      ) : null}
+
+      <fieldset>
+        <div className="field">
+          <label htmlFor="email">Email:</label>
+          <input
+            type="email"
+            name="email"
+            id="email"
+            defaultValue=""
+            ref={register({
+              required: true,
+              pattern: {
+                value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i,
+                message: 'Enter a valid e-mail address'
+              }
+            })}
+          />
+          <ErrorMessage error={errors.email} />
+        </div>
+        <div className="field">
+          <label htmlFor="password">Password:</label>
+          <input
+            name="password"
+            id="password"
+            type="password"
+            defaultValue=""
+            ref={register({
```

Comment:
interesting. I think you may answer this as part of @cheryllaird's question above but... what does this buy us over the native browser error state handling and e.g. the `required` attribute?

Reply:
It has the same standard validation as the html5 form validation does, but can be customised to anything else you need.

Lots of validation rules will apply differently based on other bits of state or even server responses, and having the validation logic inside React functions allows for this.